### PR TITLE
Display "Forgot you codename?" codename-hint link only on initial login

### LINKED ISSUE #5101: Codename hint link visible in flash message but codename hint not displayed
## Description

In the Source Interface, if a source generates a codename and proceeds to `/lookup` but elects not to submit anything at this time, the next time the source logs in and submits a message or document, the flash message includes the "Forgot your codename?" link but the `codename-hint` element of the page is missing so the link does not work.

## Steps to Reproduce

1. Run `make dev`.
1. Access the Source Interface and click "GET STARTED".
1. Take note of the codename on `/generate` and click "SUBMIT DOCUMENTS".
1. On `/lookup`, click "LOGOUT".
1. On `/index` click "LOG IN".
1. Enter the codename noted in 3 and click "CONTINUE".
1. On `/lookup` enter a message and click "SUBMIT".

## Expected Behavior

The "Success!" flash message for a first submission is displayed, **without the "Forgot your codename" link**.

## Actual Behavior

The "Success!" flash message for a first submission is displayed, **with the "Forgot your codename?" link** but the `codename-hint` element of the page is missing and therefore the link does not trigger anything.

image

## Comments

Suggestions to fix: Use the `'new_user'` session key with an `if` statement in the template to determine whether the "Forgot your codename?" link should be displayed.

This could be combined with the solution to #4401 which already involves some refactoring of the submission confirmation flash messages.

### PR DESCRIPTION
**Pull request by DrGFreeman**
## Status

Ready for review

## Description of Changes

Fixes #5101.

Changes proposed in this pull request:

Display the "Forgot you codename?" link to the `codename-hint` section in the `/submit` flashed message only on the source's initial login (i.e. do not display the link after the source logged in using their codename).

Add functional test to verify the above behavior. Also include verification of the first submission specific message.

## Testing

**Verify link does not show on second login - from `/logout`**
1. Access the Source Interface and click "GET STARTED".
1. Take note of the codename on `/generate` and click "SUBMIT DOCUMENTS".
1. On `/lookup`, click "LOGOUT".
1. On `/logout` click "LOG IN".
1. Enter the codename noted in 2 and click "CONTINUE".
1. On `/lookup` enter a message and click "SUBMIT".
    - [ ] Verify that the flashed message **does not include** the "Forgot you codename?" link
    image

**Verify link does not show on second login - from `/`**
1. Access the Source Interface and click "GET STARTED".
1. Take note of the codename on `/generate` and click "SUBMIT DOCUMENTS".
1. On `/lookup`, click "LOGOUT".
1. Click on the SecureDrop logo to return to `/`.
1. On `/` click "LOG IN".
1. Enter the codename noted in 2 and click "CONTINUE".
1. On `/lookup` enter a message and click "SUBMIT".
    - [ ] Verify that the flashed message **does not include** the "Forgot you codename?" link.
    image

**Verify link shows on first login**
1. Access the Source Interface and click "GET STARTED".
1. Take note of the codename on `/generate` and click "SUBMIT DOCUMENTS".
1. On `/lookup` enter a message and click "SUBMIT".
    - [ ] Verify that the flashed message **includes** the "Forgot you codename?" link.
    image

## Deployment

Any special considerations for deployment? Consider both:

1. Upgrading existing production instances.
2. New installs.

N/A.

### If you made changes to the server application code:

- [x] Linting (`make lint`) and tests (`make test`) pass in the development container

### If you made changes to `securedrop-admin`:

- [ ] Linting and tests (`make -C admin test`) pass in the admin development container

### If you made changes to the system configuration:

- [ ] [Configuration tests](https://docs.securedrop.org/en/latest/development/testing_configuration_tests.html) pass

### If you added or removed a file deployed with the application:

- [ ] I have updated AppArmor rules to include the change

### If you made non-trivial code changes:

- [x] I have written a test plan and validated it for this PR

Choose one of the following:

- [ ] I have opened a PR in the [docs repo](https://github.com/freedomofpress/securedrop-docs) for these changes, or will do so later
- [ ] I would appreciate help with the documentation
- [x] These changes do not require documentation

### If you added or updated a production code dependency:

Production code dependencies are defined in:

- `admin/requirements.in`
- `admin/requirements-ansible.in`
- `securedrop/requirements/python3/securedrop-app-code-requirements.in`

If you changed another `requirements.in` file that applies only to development
or testing environments, then no diff review is required, and you can skip
(remove) this section.

Choose one of the following:

- [ ] I have performed a diff review and pasted the contents to [the packaging wiki](https://github.com/freedomofpress/securedrop-debian-packaging/wiki)
- [ ] I would like someone else to do the diff review

### CODE REVIEW

**Comment by DrGFreeman** [comment]
Added functional tests, rebased to develop and set status to "Ready for review".

**Comment by DrGFreeman** [comment]
Rebased to **develop** following version 2.1.0 release.

**Comment by conorsch** [comment]
Hey there, thanks @DrGFreeman! I believe @creviera expressed an interest in reviewing this, so I'm tagging her in here. Likely next week sometime, given our current schedule.

File: securedrop/source_templates/first_submission_flashed_message.html [hunk 1]
```diff
@@ -1,8 +1,10 @@
 <section class="message">
   <h2>{{ gettext('Success!') }}</h2>
   <p>{{ gettext('Thank you for sending this information to us. Please check back later for replies.') }}
+    {% if new_user_codename %}
     <a href="#codename-hint">
       {{ gettext('Forgot your codename?') }}
```
Comment (sssoleileraaa) [inline securedrop/source_templates/first_submission_flashed_message.html]: [nit] this would look better on a new line

**Review by sssoleileraaa — DISMISSED** [review]
works as expected, just one nitpicky comment, so i'll let this sit for a few after approving

## Testing

**Verify link does not show on second login - from `/logout`**
1. Access the Source Interface and click "GET STARTED".
1. Take note of the codename on `/generate` and click "SUBMIT DOCUMENTS".
1. On `/lookup`, click "LOGOUT".
1. On `/logout` click "LOG IN".
1. Enter the codename noted in 2 and click "CONTINUE".
1. On `/lookup` enter a message and click "SUBMIT".
    - [x] Verify that the flashed message **does not include** the "Forgot you codename?" link
    image


**Verify link does not show on second login - from `/`**
1. Access the Source Interface and click "GET STARTED".
1. Take note of the codename on `/generate` and click "SUBMIT DOCUMENTS".
1. On `/lookup`, click "LOGOUT".
1. Click on the SecureDrop logo to return to `/`.
1. On `/` click "LOG IN".
1. Enter the codename noted in 2 and click "CONTINUE".
1. On `/lookup` enter a message and click "SUBMIT".
    - [x] Verify that the flashed message **does not include** the "Forgot you codename?" link.
    image

**Verify link shows on first login**
1. Access the Source Interface and click "GET STARTED".
1. Take note of the codename on `/generate` and click "SUBMIT DOCUMENTS".
1. On `/lookup` enter a message and click "SUBMIT".
    - [x] Verify that the flashed message **includes** the "Forgot you codename?" link.
    image

File: securedrop/source_templates/first_submission_flashed_message.html (hunk 1 above)
Comment (DrGFreeman) [reply securedrop/source_templates/first_submission_flashed_message.html]: Hi @creviera, thanks for the review. Just to be sure, you mean moving the "Forgot your codename?" item on a separate line like this?

image

Comment (sssoleileraaa) [reply securedrop/source_templates/first_submission_flashed_message.html]: yup, looks great!

**Comment by DrGFreeman** [comment]
Moved "Forgot your codename?" text to new line (screenshot below) as per @creviera's suggestion & rebased to develop.

image

**Review by sssoleileraaa — APPROVED** [review]
lgtm! i also created a separate but related follow-up issue (needs discussion): https://github.com/freedomofpress/securedrop/issues/6155

**Pull request merged**